client/src/components/Payment/index.js

```diff
@@ -1,29 +1,102 @@
 import React, {Component} from 'react';
 import { Container, Collapse, Button, CardBody, Card, Form, FormGroup, Label, Input } from 'reactstrap';
+import axios from 'axios';
+
+import CardInformation from '../CardInformation';
 
 export default class Payment extends Component {
     constructor(props) {
         super(props);
         this.state = {
             collapseCard: false,
             collapseAddr: false,
-            savedCards: 'No card saved',
             savedAddresses: 'No addresses saved',
+            ccName: '',
+            ccCardType: '',
+            ccCardNum: '',
+            ccExpMonth: '',
+            ccExpYear: '',
+            ccCVV: '',
+            savedCards: [],
         }
         this.toggleCard = this.toggleCard.bind(this);
```

Comment:
Instead of binding every function, when declaring the `toggleCard` function you can do this:

```javascript
toggleCard = ()=>this.setState({ collapseCard: !this.state.collapseCard });
```

This is due to the [class properties proposal](https://github.com/tc39/proposal-class-fields) and [its babel plugin](https://babeljs.io/docs/en/next/babel-plugin-proposal-class-properties) that [comes with `create-react-app`](https://reactjs.org/docs/handling-events.html).